### PR TITLE
chore: add auth-service snippets

### PR DESCRIPTION
**Pull request by kleyow**
Decided to do auth-service as well since it gives me a better view of the api definition as a whole.
No `/docs` file generated yet. Need to merge thirdparty-api-adapter changes first so I can make changes in order.

### CODE REVIEW

File: thirdparty/openapi3/schemas/AuthScopesEnum.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+title: AuthScopesEnum
```
Comment (kleyow) [inline thirdparty/openapi3/schemas/AuthScopesEnum.yaml:1]: Should we change this to `AuthScopesTypes` to keep in line with mojaloop definition standard. From what I understand Enums are denoted with `...Types`.

File: thirdparty/openapi3/schemas/ConsentRevokedAt.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+title: ConsentRevokedAt
```
Comment (kleyow) [inline thirdparty/openapi3/schemas/ConsentRevokedAt.yaml:1]: Thoughts on removing this and just using https://github.com/mojaloop/api-snippets/blob/master/v1.0/openapi3/schemas/DateTime.yaml .

i.e for `Consent`
```
  revokedAt:
    $ref: '../../../v1.0/openapi3/schemas/Datetime.yaml'
```

File: thirdparty/openapi3/schemas/ConsentStatusEnum.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+title: ConsentStatusEnum
```
Comment (kleyow) [inline thirdparty/openapi3/schemas/ConsentStatusEnum.yaml:1]: re: Enum naming scheme.
suggestion ConsentStatusEnum -> ConsentStatusType

File: thirdparty/openapi3/schemas/Credential.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+title: Credential
+type: object
+description: |
+  A credential used to allow a user to prove their identity and
+  access to an account with a DFSP.
+properties:
+  id:
+    type: string
```
Comment (kleyow) [inline thirdparty/openapi3/schemas/Credential.yaml:8]: question for auth team: Is this supposed to be a `CorrelationId`

File: thirdparty/openapi3/schemas/CredentialStatusEnum.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+title: CredentialStatusEnum
```
Comment (kleyow) [inline thirdparty/openapi3/schemas/CredentialStatusEnum.yaml:1]: re: Enum naming scheme.
suggestion CredentialStatusEnum -> CredentialStatusType

File: thirdparty/openapi3/schemas/CredentialTypeEnum.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+title: CredentialTypeEnum
```
Comment (kleyow) [inline thirdparty/openapi3/schemas/CredentialTypeEnum.yaml:1]: re: Enum naming scheme.
suggestion CredentialTypeEnum -> CredentialType

File: thirdparty/openapi3/schemas/UpdateCredentialRequest.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+description: |
+  A credential used to allow a user to prove their identity and access
+  to an account with a DFSP.
+
+  UpdateCredentialRequest is a special formatting of the credential to allow us to be
+  more explict about the `status` field - it should only ever be PENDING when updating
+  a credential.
+properties:
+  id:
+    type: string
```
Comment (kleyow) [inline thirdparty/openapi3/schemas/UpdateCredentialRequest.yaml:12]: question: is this a `CorrelationId`?

File: thirdparty/openapi3/schemas/AuthScopesEnum.yaml (hunk 1 above)
Comment (lewisdaly) [reply thirdparty/openapi3/schemas/AuthScopesEnum.yaml:1]: Yes please - if that's what we do elsewhere

File: thirdparty/openapi3/schemas/GenerateChallengeRequest.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+title: GenerateChallengeRequest
```
Comment (lewisdaly) [inline thirdparty/openapi3/schemas/GenerateChallengeRequest.yaml:1]: Should this be be `ConsentGenerateChallengePostRequest`?

File: thirdparty/openapi3/schemas/UpdateConsentRequest.yaml [hunk 9]
```diff
@@ -0,0 +1,28 @@
+title: UpdateConsentRequest
```
Comment (lewisdaly) [inline thirdparty/openapi3/schemas/UpdateConsentRequest.yaml:1]: Should this be `ConsentIdPutRequest`?

File: thirdparty/openapi3/schemas/UpdateCredentialRequest.yaml (hunk 7 above)
Comment (lewisdaly) [reply thirdparty/openapi3/schemas/UpdateCredentialRequest.yaml:12]: Nah I don't think so - see other comment about the `credential.id`

File: thirdparty/openapi3/schemas/Credential.yaml (hunk 4 above)
Comment (lewisdaly) [reply thirdparty/openapi3/schemas/Credential.yaml:8]: Nah I don't think so - this is the `KeyHandleId` which comes back from the FIDO API on the device. I'm happy to keep it as a string.

Comment (kleyow) [reply thirdparty/openapi3/schemas/Credential.yaml:8]: Thanks for clarifying.

File: thirdparty/openapi3/schemas/GenerateChallengeRequest.yaml (hunk 8 above)
Comment (kleyow) [reply thirdparty/openapi3/schemas/GenerateChallengeRequest.yaml:1]: Seems that it should be ConsentGenerateChallengeIDPostRequest. That should be good enough for now.

File: thirdparty/openapi3/schemas/UpdateConsentRequest.yaml (hunk 9 above)
Comment (kleyow) [reply thirdparty/openapi3/schemas/UpdateConsentRequest.yaml:1]: sounds good.

**Comment by kleyow** [comment]
re: KeyHandleId. String is fine for now. Couldn't fine the format of a Key Handle but it'd be nice if we had pattern matching and a explicit `KeyHandle` definition.

File: thirdparty/openapi3/schemas/UpdateCredentialRequest.yaml [hunk 10]
```diff
@@ -0,0 +1,28 @@
+title: UpdateCredentialRequest
```
Comment (kleyow) [inline thirdparty/openapi3/schemas/UpdateCredentialRequest.yaml:1]: Really confused about this definition. It doesn't be seem to be a request but more akin to a property used in `ConsentIDPutRequest`  https://github.com/mojaloop/api-snippets/pull/17/files#diff-2b332d878ba42f73f319a12fb3fccfb4R22 .

Maybe something like CredentialProof? @Auth-Team

Comment (lewisdaly) [reply thirdparty/openapi3/schemas/UpdateCredentialRequest.yaml:1]: I think this was my fault - and there's a good chance I did something wrong here. 

One of the challenges/quirks with the api design is that we have a ton of fields which could be null or undefined. My goal with request bodies (or parts of request bodies) was to have a schema which would be very explicit, and thus make writing the types and code much easier.

Renaming it from `...Request` would make this less confusing - `CredentialProof` seems suitable, or perhaps `UpdatedCredential`

Comment (kleyow) [reply thirdparty/openapi3/schemas/UpdateCredentialRequest.yaml]: UpdatedCredential sounds good. done.

**Review by eoln — APPROVED** [review]
lgtm

**Pull request merged**